### PR TITLE
Improve preview iframe

### PR DESCRIPTION
**Pull request by KutnerUri**
## Proposed Changes

- docs - move theme up the dom, and improve load times
- pubsub - replace mutation observer with a react hook

### CODE REVIEW

File: scopes/react/react/docs/base.tsx [hunk 1]
```diff
@@ -16,39 +12,14 @@ import { ComponentOverview } from './component-overview';
 import { CompositionsSummary } from './compositions-summary/compositions-summary';
 import { ExamplesOverview } from './examples-overview';
 import { Properties } from './properties/properties';
+import { useComponentDocs } from './use-component-docs';
 
 export type DocsSectionProps = {
   docs?: docsFile;
   compositions: React.ComponentType[];
   componentId: string;
 } & HTMLAttributes<HTMLDivElement>;
```
Comment (KutnerUri) [inline scopes/react/react/docs/base.tsx:22]: extracted to its own file

File: scopes/react/react/docs/base.tsx [hunk 2]
```diff
@@ -75,33 +45,30 @@ export function Base({ docs = defaultDocs, componentId, compositions, ...rest }:
   const Content: any = isFunction(docs.default) ? docs.default : () => null;
 
   return (
-    <ThemeContext>
```
Comment (KutnerUri) [inline scopes/react/react/docs/base.tsx:78]: loading/unloading themeContext and the icon font when gql finished made the browser work harder, and more likely to freeze. so, moved it to docs-app

Comment (KutnerUri) [reply scopes/react/react/docs/base.tsx:78]: I'm considering to move the whole gql together with ComponentOverview to a separate component. Not sure if it's ideal though

File: scopes/react/react/react.env.ts [hunk 3]
```diff
@@ -174,11 +168,8 @@ export class ReactEnv implements Environment {
   getDevServer(context: DevServerContext, targetConfig?: Configuration): DevServer {
     const defaultConfig = this.getWebpackConfig(context);
     const config = targetConfig ? webpackMerge(targetConfig as any, defaultConfig as any) : defaultConfig;
```
Comment (KutnerUri) [inline scopes/react/react/react.env.ts:170]: fun fact - docs file is already included using the preview link file, as "defaultModule" at **generate-link.ts**

Comment (KutnerUri) [reply scopes/react/react/react.env.ts:170]: tested in workspace, dev mode, and in bare scope 🐻

**Review by odedre — APPROVED** [review]

**Pull request merged**